### ISSUE TITLE
[bug] Hot Restart

### ISSUE THREAD

**Issue by zingwin** [Canardoux/tau#543]
The app uses FlutterSoundPlayer singleton to play audio. When it is hot restart app, the program will crash.

when call await playerModule.openAudioSession() in iOS, program will crash。 but It's good on Android

```
FlutterSoundPlayer playerModule = FlutterSoundPlayer();
  bool _isMute = false;
  bool _isPlayInited = false;

  static SoundManager _getInstance() {
    if (_instance == null) {
      _instance = new SoundManager._internal();
    }
    return _instance;
  }

  SoundManager._internal() {
    initAudioPlay();
  }

  Future<void> initAudioPlay() async {
    try {
      await playerModule.openAudioSession(
          focus: AudioFocus.requestFocusAndStopOthers,
          category: SessionCategory.playAndRecord);
      await playerModule.setSubscriptionDuration(Duration(milliseconds: 100));
      _isPlayInited = true;
      return Future.value(true);
    } catch(e){
      return Future.value(false);
    }
  }
```

**Comment by Larpoux** [comment]
Can we have your logs ?

**Comment by zingwin** [comment]
> Can we have your logs ?

```
flutter: FS:---> openAudioSession
Lost connection to device.
```

not too mush log
maybe  iOS can not  reset audio session

**Comment by Abacaxi-Nelson** [comment]
Get too this error when working on my phone and making some hot restart.

**Comment by Larpoux** [comment]
Several Flutter Sound users complained about that.

There is really something wrong in Flutter Sound. @bsutton investigated this and found the solution but he does not want to work anymore with us 👎 .

Registered in [the Flutter Sound Project](https://github.com/dooboolab/flutter_sound/projects/3)

**Comment by zxl777** [comment]
The problem should be here

`
  FlutterSoundPlayerCallback getSession(int slotno)
  {
    return _slots[slotno];
  }

`

![image](https://user-images.githubusercontent.com/816448/106731231-381d7600-664a-11eb-8520-e27d48838648.png)

![image](https://user-images.githubusercontent.com/816448/106731351-571c0800-664a-11eb-902c-ce87a886464f.png)

**Comment by Larpoux** [comment]
Yes, someone told me that the slot manager does not work well with hot restart.
This must be investigated ... and fixed.

**Comment by zxl777** [comment]
This bug should be solved first. Every hot restart causes the crash to disconnect, and it takes at least 3 minutes to restart the app. Greatly affect development efficiency.

**Comment by zxl777** [comment]
In Hot Restart on iOS, it will still cause the app to crash and disconnect the debug connection.
I now have to develop on Android first to avoid encountering this problem frequently.

Hope someone can solve this problem.

**Comment by Larpoux** [comment]
Fixed in Flutter Sound release 7.6.3
